dc-cudami-server/dc-cudami-server-backend-jdbi/src/main/java/de/digitalcollections/cudami/server/backend/impl/jdbi/JdbiRepositoryImpl.java

```diff
@@ -34,4 +86,225 @@ public String getTableAlias() {
   public String getTableName() {
     return tableName;
   }
+
+  protected String getWhereClause(
+      FilterCriterion<?> fc, Map<String, Object> argumentMappings, int criterionCount)
+      throws IllegalArgumentException, UnsupportedOperationException {
+    StringBuilder query = new StringBuilder();
+    if (fc != null) {
+      FilterOperation filterOperation = fc.getOperation();
+      // @see https://www.postgresql.org/docs/11/functions.html
+      final String fieldName = fc.getFieldName();
+      final String columnName = getColumnName(fieldName);
+      final String leftSide = (columnName != null) ? columnName : fieldName;
+      // basic sql injection detection:
+      if (leftSide.contains("'") || leftSide.contains(";")) {
+        throw new IllegalArgumentException(
+            String.format("leftSide '%s' seems to contain malicious code!", leftSide));
+      }
+
+      String criterionKey = KEY_PREFIX_FILTERVALUE + criterionCount;
+      switch (filterOperation) {
+        case BETWEEN:
+          if (fc.getMinValue() == null || fc.getMaxValue() == null) {
+            throw new IllegalArgumentException("For 'BETWEEN' operation two values are expected");
+          } else {
+            // example: BETWEEN '2015-01-01' AND '2015-12-31'
+            String keyMin = criterionKey + "_min";
+            String keyMax = criterionKey + "_max";
+            query
+                .append("(")
+                .append(leftSide)
+                .append(" BETWEEN ")
+                .append(":")
+                .append(keyMin)
+                .append(" AND ")
+                .append(":")
+                .append(keyMax)
+                .append(")");
+            argumentMappings.put(keyMin, fc.getMinValue());
+            argumentMappings.put(keyMax, fc.getMaxValue());
+          }
+          break;
+        case IN:
+        case NOT_IN:
+          if (fc.getValues() == null || fc.getValues().isEmpty()) {
+            throw new IllegalArgumentException(
+                "For 'IN/NOT_IN' operation at least one value is expected");
+          }
+          // For 'in' or 'nin' operation
+          query.append("(").append(leftSide);
+          if (filterOperation == FilterOperation.NOT_IN) {
+            query.append(" NOT");
+          }
+          query.append(" IN (");
+          int valueCount = 1;
+          ArrayList<String> values = new ArrayList<>();
+          for (Object value : fc.getValues()) {
```

Comment:
normale for schleife

Reply:
ehrlich gesagt, finde ich den code nicht besser:

```
Object[] valuesArray = fc.getValues().toArray();
          for (int valueCount = 1; valueCount < valuesArray.length; valueCount++) {
            Object value = valuesArray[valueCount - 1];
            String key = criterionKey + "_" + valueCount;
            values.add(":" + key);
            argumentMappings.put(key, value);
          }
```

vorher (besser meiner meinung nach):

```
int valueCount = 1;
          for (Object value : fc.getValues()) {
            String key = criterionKey + "_" + valueCount;
            values.add(":" + key);
            argumentMappings.put(key, value);
            valueCount++;
          }
```

Reply:
replaced by stream